### PR TITLE
[MM-69810] Update golang.org/x/image dep

### PR DESCRIPTION
**Pull request by isacikgoz**
#### Summary
See the ticket.

#### Ticket Link

Fixes: https://mattermost.atlassian.net/browse/MM-69810

#### Release Note
```release-note
NONE
```

## Change Impact: 🟡 Medium

**Regression Risk:** Image decoding now enforces a shared `MaxDecodedResolution` cap (including for non-seekable inputs) and returns earlier/changed error behavior for oversized images across channels and team icon processing. Although covered by new decoder unit tests, this can still affect edge-case image inputs and error semantics.  
**QA Recommendation:** Manual QA can be skipped; rely on automated tests, with an optional quick smoke test for team icon uploads and channel image rendering.
  
*Generated by CodeRabbitAI*

### CODE REVIEW

**Review by edgarbellot — APPROVED** [review]
LGTM 👍 happy path + edge cases covered, thank you Ibrahim!

File: e2e-tests/playwright/specs/functional/channels/post_list/post_height.spec.ts [hunk 1]
```diff
@@ -277,8 +277,11 @@ test.describe('Post height', () => {
         },
         {
             name: 'post with an SVG Markdown image',
-            // TODO Either Chrome preloads the SVG's dimensions early or Firefox doesn't allocate the height properly
-            skipProjects: ['firefox'],
+            // Markdown/remote SVGs intentionally receive no server-provided dimensions
+            // (SVG images are filtered from link metadata to mitigate the MM-67372 DoS),
+            // so the client cannot reserve height before the SVG loads. That makes a
+            // layout-shift-free render impossible for this case regardless of browser.
+            skipProjects: ['chrome', 'firefox', 'ipad'],
```
Comment (hmhealey) [inline e2e-tests/playwright/specs/functional/channels/post_list/post_height.spec.ts:284]: These changes don't seem to be needed. I'm actively working on these tests, so I'm going to revert them in https://github.com/mattermost/mattermost/pull/37653

File: webapp/channels/src/components/size_aware_image.tsx [hunk 2]
```diff
@@ -408,11 +408,6 @@ export class SizeAwareImage extends React.PureComponent<Props, State> {
         } = this.props;
         const renderPlaceholderOnly = this.props.renderPlaceholderOnly ?? false;
 
-        let ariaLabelImage = this.props.intl.formatMessage({id: 'file_attachment.thumbnail', defaultMessage: 'file thumbnail'});
```
Comment (hmhealey) [inline webapp/channels/src/components/size_aware_image.tsx:411]: These changes don't seem to have anything to do with MM-69810
